gitcoins-cordapp/workflows/src/test/kotlin/com/gitcoins/GitEventFlowTests.kt

```diff
@@ -0,0 +1,113 @@
+
+    @Before
+    fun setup() {
+        network = MockNetwork(
+                cordappPackages = listOf(
+                        "com.r3.corda.sdk.token.contracts",
+                        "com.r3.corda.sdk.token.workflow",
+                        "com.gitcoins.schema"
+                ),
+                threadPerNode = true,
+                networkParameters = testNetworkParameters(minimumPlatformVersion = 4))
+
+        a = network.createNode()
+        user = a.info.singleIdentity()
+    }
+
+    @After
+    fun tearDown() {
+        network.stopNodes()
+    }
+
+    @Test
+    fun `Query git users`() {
+        val result = a.startFlow(QueryGitUserDatabaseFlow(gitUser)).getOrThrow()
+        assertThat(result).isEmpty()
+    }
+
+    @Test
+    fun `Create new public key`() {
+        val future = a.startFlow(CreateKeyFlow(gitUser))
+        network.waitQuiescent()
+        future.getOrThrow()
+
+        val result = a.startFlow(QueryGitUserDatabaseFlow(gitUser)).getOrThrow()
+        assertEquals(gitUser, result.first().gitUserName)
+
+        val duplicate = a.startFlow(CreateKeyFlow(gitUser))
+        assertFailsWith<FlowException> { duplicate.getOrThrow() }
+    }
+
+    @Test
+    fun `Issue git token from push event`() {
+        val futureFail = a.startFlow(PushEventFlow(gitUser))
+        assertFailsWith<FlowException> { futureFail.getOrThrow() }
+
+        val future = a.startFlow(CreateKeyFlow(gitUser))
+        network.waitQuiescent()
+        future.getOrThrow()
+
+        val signedTx = a.transaction { a.startFlow(PushEventFlow(gitUser)) }.getOrThrow()
+        assertEquals(signedTx, a.services.validatedTransactions.getTransaction(signedTx.id))
+
+        val token = signedTx.tx.outRefsOfType<FungibleToken<GitToken>>().single()
+        val vaultToken = a.services.vaultService.queryBy<FungibleToken<GitToken>>().states.single()
+        assertEquals(token, vaultToken)
+
+        val tokenBalance = a.services.vaultService.tokenBalance(GitToken())
+        assertEquals(1, tokenBalance.quantity)
+    }
+
+    @Test
+    fun `Issue git token from pull request review event`() {
+        val futureFail = a.startFlow(PullRequestReviewEventFlow(gitUser))
+        assertFailsWith<FlowException> { futureFail.getOrThrow() }
+
+        val future = a.startFlow(CreateKeyFlow(gitUser))
+        network.waitQuiescent()
+        future.getOrThrow()
+
+        val signedTx = a.transaction { a.startFlow(PullRequestReviewEventFlow(gitUser)) }.getOrThrow()
+        assertEquals(signedTx, a.services.validatedTransactions.getTransaction(signedTx.id))
+
+        val token = signedTx.tx.outRefsOfType<FungibleToken<GitToken>>().single()
+        val vaultToken = a.services.vaultService.queryBy<FungibleToken<GitToken>>().states.single()
+        assertEquals(token, vaultToken)
+
+        val tokenBalance = a.services.vaultService.tokenBalance(GitToken())
+        assertEquals(1, tokenBalance.quantity)
+    }
+}
```

Comment:
Add failure scenarios. I would also like to see integration test using webserver.